src/plugins/data/public/ui/dataset_selector/configurator.tsx

```diff
@@ -123,6 +158,79 @@ export const Configurator = ({
           >
             <EuiFieldText disabled value={dataset.title} />
           </EuiFormRow>
+          {indexedViewsService && (
+            <>
+              <EuiSpacer />
+              <EuiSwitch
+                compressed
+                checked={selectIndexedView}
+                label={
+                  <EuiFormLabel>
+                    {i18n.translate(
+                      'data.explorer.datasetSelector.advancedSelector.configurator.showAvailableIndexedViewsLabel',
+                      {
+                        defaultMessage: 'Query indexed view'
+                      }
+                    )}
+                  </EuiFormLabel>  }
+                onChange={(e) => setSelectIndexedView(e.target.checked)}
```

Comment:
Didn't @riysaxen-amzn add tests for a bunch of these? Looks like they're all here, but not sure why coverage report looks different